rinja_parser/src/tests.rs

```diff
@@ -1064,3 +1064,39 @@ fn fuzzed_filter_recursion() {
     const TEMPLATE: &str = include_str!("../tests/filter-recursion.txt");
     assert!(Ast::from_str(TEMPLATE, None, &Syntax::default()).is_err());
 }
+
+#[test]
+#[should_panic = "called `Result::unwrap()` on an `Err` value: \
+    \"delimiters must be at most 32 characters long. The closing block delimiter \
+    (\\\"\\\\0]***NEWFILE\\\\u{1f}***\\\"...) is too long\""]
+fn fuzzed_excessive_syntax_lengths() {
+    let syntax = SyntaxBuilder {
+        name: "test",
+        block_start: Some("\0*DD\0\0"),
+        block_end: Some(
+            "\0]***NEWFILE\u{1f}***:7/v/.-3/\u{1b}/~~~~z~0/*:7/v/./t/t/.p//NEWVILE**::7/v",
+        ),
+        expr_start: Some("/."),
+        expr_end: None,
+        comment_start: Some(include_str!(
+            "../tests/excessive_syntax_lengths-comment-start.bin"
+        )),
+        comment_end: None,
+    }
+    .to_syntax()
+    .unwrap();
+
+    let (sender, receiver) = std::sync::mpsc::channel();
```

Comment:
Same here: don't you want to split it into its own test? I think this code is meant to ensure that if the previous code doesn't panic, it'll not time out the test. If so, please add a comment.